Composer/packages/client/src/utils/fileUtil.ts

```diff
@@ -6,19 +6,19 @@ import moment from 'moment';
 import { FileTypes, SupportedFileTypes } from '../constants';
 import { File } from '../store/types';
 
-export function getExtension(filename?: string): string | any {
-  if (typeof filename !== 'string') return filename;
-  return filename.substring(filename.lastIndexOf('.') + 1, filename.length) || filename;
+export function getExtension(filename: any): string {
+  if (typeof filename === 'string') return filename.substring(filename.lastIndexOf('.') + 1, filename.length);
+  return filename;
 }
 
-export function getBaseName(filename?: string): string | any {
-  if (typeof filename !== 'string') return filename;
-  return filename.substring(0, filename.lastIndexOf('.')) || filename;
+export function getBaseName(filename: any): string {
+  if (typeof filename === 'string') return filename.substring(0, filename.lastIndexOf('.'));
+  return filename;
 }
 
-export function upperCaseName(filename?: string): string | any {
-  if (typeof filename !== 'string') return filename;
-  return filename.charAt(0).toUpperCase() + filename.slice(1);
+export function upperCaseName(filename: any): string {
+  if (typeof filename === 'string') return filename.charAt(0).toUpperCase() + filename.slice(1);
```

Comment:
I personally like the syntax isString(filename). We leverage the entire lodash library. We could use it to have a cleaner check

Reply:
I actually prefer using primitives where it makes sense. No need to hide some pretty basic syntax behind a utility function.

That being said, I am still a fan of lodash for other, more complex, operations (map, reduce, etc) because they often work on 'collections` as opposed to just arrays.

Reply:
Yah. It was just a personal preference for me. I was writing apps that required support for older browsers. By just practice I was using all lodash functions if available to avoid things breaking. But we could go either way on typeof checks as they are primitive